extensions/shared/external-media/sources/google-photos/filter-option.js

```diff
@@ -12,8 +18,21 @@ import {
 	GOOGLE_PHOTOS_CATEGORIES,
 	GOOGLE_PHOTOS_DATE_PRESETS,
 	DATE_RANGE_ANY,
+	DATE_RANGE_CUSTOM,
+	MONTH_SELECT_OPTIONS,
+	CURRENT_YEAR,
 } from '../../constants';
 
+/**
+ * This uses the experimental NumberControl from the block editor where available,
+ * otherwise it falls back to a standard TextControl, limited to numbers.
+ */
+const NumberControl =
+	BlockEditorNumberControl ||
+	function CustomNumberControl( props ) {
```

Comment:
> simple question: are we safe using this experimental component across different core versions?

It answers my question 😅 

I think we should use `typeof` here:

```suggestion
const NumberControl = typeof BlockEditorNumberControl !== 'undefined'
		? BlockEditorNumberControl ||
		: function CustomNumberControl( props ) {
```

Reply:
@retrofox Is using `typeof` just a preference or would we run into bugs if it was left as is?

Reply:
It's a preference.